app/src/main/java/org/thoughtcrime/securesms/preferences/SettingsActivity.kt

```diff
@@ -228,23 +229,65 @@ class SettingsActivity : PassphraseRequiredActionBarActivity() {
         }
     }
 
+    // Method to update the display name if possible
+    private fun handleDisplayNameChange(displayName: String?) {
+
+        // If we don't have a working network connection we'll decline the display name change and bail..
+        val haveNetworkConnection = NetworkUtils.haveValidNetworkConnection(this@SettingsActivity)
+        if (!haveNetworkConnection) {
+            binding.loader.isVisible = false
+            Log.w(TAG, "Declining attempt to change display name without network connection.")
+            Toast.makeText(this@SettingsActivity, R.string.profileErrorUpdate, Toast.LENGTH_LONG).show()
+            return
+        }
+
+        // ..otherwise we'll attempt to make the change.
+        var failedToUpdateDisplayName = false
+
+        // As long as we have a new display name to set..
+        if (displayName != null) {
+            TextSecurePreferences.setProfileName(this, displayName)
+            val user = configFactory.user
+            if (user == null) {
+                // Couldn't get the user for some reason, so we'll fail to change the display name
+                Log.w(TAG, "Could not obtain user details from configFactory - failing display name change.")
+                failedToUpdateDisplayName = true
+            } else {
+                configFactory.user?.setName(displayName)
+            }
+        } else {
+            // If for some reason we were asked to set a null display name we'll not..
+            failedToUpdateDisplayName = true
+            Log.w(TAG, "Asked to set display name to null - not doing that.")
+        }
+
+        // Regardless of success or failure we'll hide the loader animation (three animated dots)
+        binding.loader.isVisible = false
+
+        // Either warn the user there was an issue or set the new display name as appropriate
+        if (failedToUpdateDisplayName) {
+            Toast.makeText(this@SettingsActivity, R.string.profileErrorUpdate, Toast.LENGTH_LONG).show()
+        } else {
+            binding.btnGroupNameDisplay.text = displayName
+        }
```

Comment:
Is any of the code above even required if displayName is null?
Shouldn't all of this be reorganised and moved into updateProfile?

Maybe in updateProfile you'd have:
1) Check for the network first and show error if not
2) if displayName is not null { do what you need there }
3) if isUpdatingProfilPicture { do what you need  in there }
4) do some clean ups / setting some views visible or hidden appropriately

Wouldn't that work?

Reply:
Yeah, I can do that. I had separated out the display name handling from `updateProfile` because I felt the method was getting a little bit bloated, and usually you try to write methods that 'do one thing' - but I guess in this case that 'one thing' can be "handle profile updates", which encompasses both the display name and profile picture.

Reply:
You can still have it in separate functions, it's more a logical / structural change.
The parent function should maybe organise the overall structure in a way that makes more sense. Then the chinkier parts of the code can be extracted in different methods if that is better.

Reply:
Have done some refactoring / tidying and tested that it all works (it does) - let me know if you'd like anything further done.